templates/consonant/consonant.js

```diff
@@ -0,0 +1,1047 @@
+  const resp = await fetch(`${getRootPath()}/placeholders.json`);
+  const json = await resp.json();
+  const placeholders = {};
+  json.data.forEach((placeholder) => {
+    placeholders[placeholder.Key] = placeholder.Text;
+  });
+  return (placeholders);
+}
+
+/**
+ * loads a script by adding a script tag to the head.
+ * @param {string} url URL of the js file
+ * @param {Function} callback callback on load
+ * @param {string} type type attribute of script tag
+ * @returns {Element} script element
+ */
+
+export function loadScript(url, callback, type) {
+  const head = document.querySelector('head');
+  const script = document.createElement('script');
+  script.setAttribute('src', url);
+  if (type) {
+    script.setAttribute('type', type);
+  }
+  head.append(script);
+  script.onload = callback;
+  return script;
+}
+
+/**
+ * loads everything needed to get to LCP.
+ */
+async function loadEager() {
+  const main = document.querySelector('main');
+  if (main) {
+    decorateMain(main);
+    document.querySelector('body').classList.add('appear');
+    const lcpBlocks = ['featured-article', 'article-header'];
+    const block = document.querySelector('.block');
+    const hasLCPBlock = (block && lcpBlocks.includes(block.getAttribute('data-block-name')));
+    if (hasLCPBlock) await loadBlock(block, true);
+    const lcpCandidate = document.querySelector('main img');
+    const loaded = {
+      then: (resolve) => {
+        if (lcpCandidate && !lcpCandidate.complete) {
+          lcpCandidate.addEventListener('load', () => resolve());
+          lcpCandidate.addEventListener('error', () => resolve());
+        } else {
+          resolve();
+        }
+      },
+    };
+    await loaded;
+  }
+}
+
+/**
+ * loads everything that doesn't need to be delayed.
+ */
+async function loadLazy() {
+  const main = document.querySelector('main');
+
+  // post LCP actions go here
+  sampleRUM('lcp');
+
+  /* load footer */
+  const footer = document.querySelector('footer');
+  footer.setAttribute('data-block-name', 'footer');
+  footer.setAttribute('data-footer-source', `${getRootPath()}/footer`);
+  loadBlock(footer);
+
+  loadBlocks(main);
+  loadCSS('/templates/consonant/styles/lazy-styles.css');
+  addFavIcon('/templates/consonant/styles/favicon.svg');
+}
+
+/**
+ * loads everything that happens a lot later, without impacting
+ * the user experience.
+ */
+function loadDelayed() {
+  /* trigger delayed.js load */
+  const delayedScript = '/templates/consonant/scripts/delayed.js';
+  const usp = new URLSearchParams(window.location.search);
+  const delayed = usp.get('delayed');
+
+  if (!(delayed === 'off' || document.querySelector(`head script[src="${delayedScript}"]`))) {
+    let ms = 3500;
+    const delay = usp.get('delay');
+    if (delay) ms = +delay;
+    setTimeout(() => {
+      loadScript(delayedScript, null, 'module');
+    }, ms);
+  }
+}
+
+function loadMartech() {
+  const env = getHelixEnv();
```

Comment:
i think it would be great to be able to turn off the loading of martech as i don't think you can reach the LH of `100` with the normal martech stack. also it may very well be that this should actually load the stock.adobe.com martech stack / report suites, so the instrumentation could be different altogether.

so i would probably put something in there that checks for a query param in the form of `?martech=off` and if that's present returns immediately from the `loadMartech()` function.

Reply:
OK This should be fixed now, thank you for the feedback. It doesn't seem to have changed the LH score much though...